instructionAPI/src/Ternary.C

```diff
@@ -0,0 +1,131 @@
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include "Ternary.h"
+#include "../../common/src/Singleton.h"
+#include <vector>
+#include <set>
+#include <sstream>
+#include "Visitor.h"
+#include "../../common/src/singleton_object_pool.h"
+#include "InstructionDecoder-power.h"
+#include "dyn_regs.h"
+#include "ArchSpecificFormatters.h"
+
+using namespace std;
+
+extern bool ia32_is_mode_64();
+
+
+namespace Dyninst
+{
+  namespace InstructionAPI
+  {
+    TernaryAST::TernaryAST(Expression::Ptr c, Expression::Ptr f, Expression::Ptr s , Result_Type result_type):
+        Expression(result_type) , cond(c) , first(f) , second(s)
+    {
+    }
+
+
+    TernaryAST::~TernaryAST()
+    {
+    }
+    void TernaryAST::getChildren(vector<InstructionAST::Ptr>& /*children*/) const
+    {
+      return;
+    }
+    void TernaryAST::getChildren(vector<Expression::Ptr>& /*children*/) const
+    {
+        return;
+    }
+    void TernaryAST::getUses(set<InstructionAST::Ptr>& uses)
+    {
+        uses.insert(shared_from_this());
+        return;
+    }
+    bool TernaryAST::isUsed(InstructionAST::Ptr findMe) const
+    {
+        return false; //TODO
+        //return findMe->checkRegID(m_Reg, m_Low, m_High);
+    }
+
+    std::string TernaryAST::format(Architecture arch, formatStyle f) const
+    {
+        return TernaryAST::format(f); // TODO
+        //return ArchSpecificFormatter::getFormatter(arch).formatTernary(m_Reg.name());
+    }
+
+    std::string TernaryAST::format(formatStyle) const
+    {
+        std::string name = "("+cond->format() +"?" + first->format() + ":" + second->format()+ ")";
+        
+        /*std::string name = m_Reg.name();
+        std::string::size_type substr = name.rfind(':');
+        if(substr != std::string::npos)
+        {
+            name = name.substr(substr + 1, name.length());
+        }*/
+
+        /* we have moved to AT&T syntax (lowercase registers) */
+        std::transform(name.begin(), name.end(), name.begin(), ::toupper);
+        return name;
+    }
+    
+    bool TernaryAST::operator<(const TernaryAST& rhs) const
+    {
+        return false;
+    }
+    bool TernaryAST::isStrictEqual(const InstructionAST& rhs) const
+    {
+          return false;
+    }
+
+    void TernaryAST::apply(Visitor* v)
+    {
+        //v->visit(this); // TODO need to support this in visitor
+    }
+    bool TernaryAST::bind(Expression* e, const Result& val)
+    {
+        return false; // false
+        if(Expression::bind(e, val)) {
```

Comment:
Unreachable code.

Reply:
This part I marked it as TODO as I'm not sure how should we bind an Ternary AST yet.